src/shared/ui/layout/InfiniteScroll.tsx

```
@@ -0,0 +1,54 @@
+    observer.observe(sentinel);
+    return () => observer.disconnect();
+  }, [hasMore, loading, onLoadMore, threshold]);
+
+  return (
+    <div className={cn('flex flex-col', className)}>
+      {children}
+      <div ref={sentinelRef} className="shrink-0" />
+      {loading && (
+        <div className="flex items-center justify-center py-4">
+          {loader ?? (
+            <div className="w-5 h-5 border-2 border-accent-500/30 border-t-accent-500 rounded-full animate-spin" />
```

ESLint

[error] Use LoadingSpinner component instead of inline animate-spin classes

(no-restricted-syntax)